File: app/src/main/java/com/theappsolutions/boilerplate/ui/base/BaseView.java
package com.theappsolutions.boilerplate.ui.base;

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
public interface BaseView {

    void showToastMessage(String text);

    void showToastMessage(int textRes);

    void goBack();

    void close();

    void showProgress();

    void dismissProgress();
}

File: app/src/main/java/com/theappsolutions/boilerplate/ui/base/BaseView.kt
package com.theappsolutions.boilerplate.ui.base

/**
 * @author Severyn Parkhomenko s.parkhomenko@theappsolutions.com
 * @copyright (c) 2018 TheAppSolutions. (https://theappsolutions.com)
 */
interface BaseView {

    fun showToastMessage(text: String)

    fun showToastMessage(textRes: Int)

    fun goBack()

    fun close()

    fun showProgress()

    fun dismissProgress()
}
